cpp/modmesh/python/wrapper/modmesh/wrap_SimpleArray.cpp

```diff
@@ -163,99 +150,308 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
             ;
     }
 
+    static void setitem_parser(wrapped_type & arr_out, pybind11::args & args)
+    {
+        namespace py = pybind11;
+
+        if (args.size() == 2)
+        {
+            // sarr[K] = V
+            if (py::isinstance<py::int_>(args[0]) &&
+                (py::isinstance<py::bool_>(args[1]) || py::isinstance<py::int_>(args[1]) ||
```

Comment:
It seems this is just `!isinstance<py::array>` and does not need to use a list of types.

Reply:
I presume an exception is thrown with `sarr[K] = (1, 2, 3)` when the check is only `!isinstance<py::array>`.  Am I right?

Reply:
you are right!
    sarr[0, 0, 0, 0, 0] = (1, 2, 4)
RuntimeError: Unable to cast Python instance of type <class 'tuple'> to C++ type 'double'

Reply:
That exception seems to come from pybind11.  Using that exception is more maintainable than explicitly checking for Python type inspection.